include/swift/Syntax/SyntaxKind.h.gyb

```diff
@@ -71,6 +71,8 @@ bool isTokenKind(SyntaxKind Kind);
 
 bool isUnknownKind(SyntaxKind Kind);
 
+bool shallBeOmittedWhenNoChildren(SyntaxKind Kind);
```

Comment:
I didn't want to create a gyb file in the parser for this. It is currently still nice and gyb-free.

Maybe we could add this function to the `swift` namespace instead of the `swift::syntax` one and call it `parserShallOmitIfNoChildern`

Reply:
yeah, it works too, as long as we make it explicit that it's a parser thing rather than for syntax nodes in general.